extensions/http-curl/client/HTTPClient.h

```diff
@@ -297,6 +301,8 @@ class HTTPClient : public BaseHTTPClient, public core::Connectable {
 
   std::chrono::milliseconds keep_alive_idle_{-1};
 
+  std::optional<std::pair<std::string, std::string>> username_password_;
```

Comment:
Use a proper named type over a tuple. Something like `struct BasicAuthCredentials { std::string username; std::string password; };`

Reply:
yeah it does help the readability changed it in https://github.com/apache/nifi-minifi-cpp/pull/1349/commits/fdb1bf2a34423815cfe2cb395e83dcee3bc2f1cd#diff-475912c726caf957fa46ce1b55d3e845e89aa19b4c7caf2cd4a0ae1f860cbdd5R304-R311